python/grass/jupyter/timeseriesmap.py

```diff
@@ -297,30 +302,29 @@ def render(self):
                 "TimeSeriesMap.add_vector_series() to add dataset"
             )
 
-        # Create name for empty layers
-        # Random name needed to avoid potential conflict with layer names
-        # A new random_name_none is created each time the render function is run,
-        # and any existing random_name_none file will be ignored
         random_name_none = gs.append_random("none", 8) + ".png"
 
-        # Render each layer
+        tasks = []
         for date, layer in self._date_layer_dict.items():
             if layer == "None":
-                # Create file
                 filename = os.path.join(self._tmpdir.name, random_name_none)
-                self._base_filename_dict[date] = filename
-                # Render blank layer if it hasn't been done already
-                if not os.path.exists(filename):
-                    shutil.copyfile(self.base_file, filename)
-                    self._render_blank_layer(filename)
             else:
-                # Create file
                 filename = os.path.join(self._tmpdir.name, f"{layer}.png")
-                # Copying the base_file ensures that previous results are overwritten
-                shutil.copyfile(self.base_file, filename)
-                self._base_filename_dict[date] = filename
-                # Render image
-                self._render_layer(layer, filename)
+            tasks.append((date, layer, filename))
+
+        nprocs_env = gs.gisenv().get("NPROCS")
+        num_images = len(self._date_layer_dict)
+        if nprocs_env is not None:
+            num_cores = int(nprocs_env)
+        else:
+            num_cores = min(num_images, max(1, multiprocessing.cpu_count() - 1))
```

Comment:
First resolve the number of available cores (see the comment about `sched_getaffinity`) and then resolve how many we request given `NPROCS` and number of tasks. I think we probably should take `NPROCS` as a suggestion and limit the number by the number of available cores.

Reply:
From the previous comments, I understood that env vars should have priority. You think they should be a suggestion instead? If a number of cores is explicitly passed, why not respect it instead of trying to be smart about it?